src/hil/src/spi.rs

```diff
@@ -108,7 +112,27 @@ pub trait SpiMaster {
     // is any of the read/read_write calls. These functions
     // allow an application to manually control when the
     // CS line is high or low, such that it can issue multi-byte
-    // requests with single byte oprations.
+    // requests with single byte operations.
     fn hold_low(&self);
     fn release_low(&self);
 }
+
+/// SPIMasterDevice provides a chip-specific interface to the SPI Master
+/// hardware. The interface wraps the chip select line so that chip drivers
+/// cannot communicate with different SPI devices.
+pub trait SPIMasterDevice {
```

Comment:
This is nit-picky, but I wonder if `SPIDeviceMaster` is a better name. Basically, this trait is represents a single (virtualized) SPI device, and it's the master side of that device.

Alternatively, maybe `SPIDevice` is descriptive enough? There isn't really an equivalent for the slave side, is there?

I realize this is following the example in `virtual_i2c` but in hind site, those names aren't very forward looking. Basically, I think those names should be `I2CMaster` (rather than `I2C`) and `I2CDevice` (as it is now). We'll eventually add `I2CSlave`.